alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportDeleteMonitorAction.kt

```diff
@@ -150,8 +150,8 @@ class TransportDeleteMonitorAction @Inject constructor(
             monitorId: String,
         ): DeleteResponse {
             val deleteResponse = deleteMonitorDocument(client, deleteRequest)
-            deleteMetadata(client, monitor)
             deleteDocLevelMonitorQueriesAndIndices(client, monitor, monitorId)
+            deleteMetadata(client, monitor)
```

Comment:
Lets have this fix in a separate pr. This fix is for a bug, but the PR is for a compile fix because of new breaking changes that is only for 3.0. The bug is there on 2.x too, which we would want to backport

Reply:
done